Automattic/woocommerce-payments
# Register Independent ECE Payment Methods within the Blocks Implementation

Fixes #9045

#### Changes proposed in this Pull Request

This PR registers each Express Checkout Element (ECE) payment method individually in WC Blocks, allowing the Blocks API to manage these payment methods separately.

#### Testing instructions

**Pre-requisites:**

- The ECE feature flag is enabled (`npm run wp option update _wcpay_feature_stripe_ece 1`).
- Confirm that Apple Pay and Google Pay are enabled in the merchant settings.
- Use Safari on an Apple Pay-enabled site to view both buttons (Apple Pay and GooglePay).

**Test the WC Blocks ECE implementation:**

1. Add a product to the cart.
2. Navigate to the Blocks cart page.
3. Verify that both buttons are visible.
4. Complete the purchase using Apple Pay or Google Pay, then repeat the process with the other payment method to confirm both buttons function correctly.

Repeat these steps for the Blocks Checkout page.

-------------------


**Post merge**
